admin/src/pages/teams/figma-teams-connector.tsx

```diff
@@ -0,0 +1,176 @@
+import LoadingButton from '@atlaskit/button/loading-button';
+import Form, { ErrorMessage, Field, FormFooter } from '@atlaskit/form';
+import SectionMessage from '@atlaskit/section-message';
+import TextField from '@atlaskit/textfield';
+import { token } from '@atlaskit/tokens';
+import { css } from '@emotion/react';
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import { useState } from 'react';
+
+import { ConnectTeamSuccessScreen } from './connect-teams-success-screen';
+
+import { connectTeam } from '../../api';
+import { ConnectBanner, FigmaPermissionsPopup, Page } from '../../components';
+import { parseTeamIdFromFigmaUrl } from '../../utils';
+
+type ConnectTeamProps = {
+	onClose?: () => void;
+	site: string;
+};
+
+export function FigmaTeamConnector({ onClose, site }: ConnectTeamProps) {
+	const queryClient = useQueryClient();
+	const [validationError, setValidationError] = useState<string | null>(null);
+	const connectTeamMutation = useMutation({
+		mutationFn: async (teamId: string) => {
+			return (await connectTeam(teamId)).data;
+		},
+		onSuccess: () => {
+			return queryClient.invalidateQueries({ queryKey: ['teams'] });
+		},
+	});
+
+	let content: React.ReactNode = null;
+	if (connectTeamMutation.isSuccess) {
+		content = (
+			<ConnectTeamSuccessScreen
+				onClose={onClose}
+				teamName={connectTeamMutation.data.teamName}
+			/>
+		);
+	} else {
+		content = (
+			<div
+				css={css`
+					align-items: center;
+					display: flex;
+					flex-direction: column;
+					gap: 32px;
+					width: 580px;
+				`}
+			>
+				<div
+					css={css`
+						align-items: center;
+						display: flex;
+						flex-direction: column;
+						gap: 8px;
+					`}
+				>
+					<ConnectBanner />
+					<div
+						css={css`
+							color: ${token('color.text')};
+							font-size: 24px;
+							font-weight: 500;
+						`}
+					>
+						Connect Figma to Jira
+					</div>
+				</div>
+				{connectTeamMutation.isError && (
+					<ConnectTeamUnauthorizedSectionMessage />
```

Comment:
While I was testing this earlier I discovered there was an issue with the connect team use case where the issue property write was failing and resulting in a `500` from the app (should now be fixed in this [PR](https://github.com/atlassian-labs/figma-for-jira/pull/125)). It threw me a little because I still saw the unauthorized message in the UI. It can be done in separate PRs but I think we should look at returning specific error codes / messages from some of these use cases so we can display more accurate error messages in the UI

Reply:
agreed! This is what i was referring to in:
> I'm not sure if I need to be added to a feature flag for this to work but I was getting 415 errors with await jiraService.setConfigurationStateInAppProperties() which was recently added to connect-figma-team-use-case.ts. I just commented that part out when testing this flow

and

> Upon error, right now it will show the permissions error. We should update the /admin/teams/:team_id/connect route to return a 403 on when it fails to connect the team due to permissions errors and 500 for other errors when appropriate.